src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbilityCooldown.kt

```diff
@@ -4,272 +4,259 @@ import at.hannibal2.skyhanni.SkyHanniMod
-                val recentItems = InventoryUtils.recentItemsInHand.values
-                if (VOODOO_DOLL_WILTED in recentItems) {
-                    ItemAbility.VOODOO_DOLL_WILTED.sound()
-                }
-            }
-            // Golem Sword
-            event.soundName == "random.explode" && event.pitch == 4.047619f && event.volume == 0.2f -> {
-                ItemAbility.GOLEM_SWORD.sound()
-            }
-            // Weird Tuba & Weirder Tuba
-            event.soundName == "mob.wolf.howl" && event.volume == 0.5f -> {
-                val recentItems = InventoryUtils.recentItemsInHand.values
-                if (WEIRD_TUBA in recentItems) {
-                    ItemAbility.WEIRD_TUBA.sound()
-                }
-                if (WEIRDER_TUBA in recentItems) {
-                    ItemAbility.WEIRDER_TUBA.sound()
-                }
-            }
-            // End Stone Sword
-            event.soundName == "mob.zombie.unfect" && event.pitch == 2.0f && event.volume == 0.3f -> {
-                ItemAbility.END_STONE_SWORD.sound()
-            }
-            // Soul Esoward
-            event.soundName == "mob.wolf.panting" && event.pitch == 1.3968254f && event.volume == 0.4f -> {
-                ItemAbility.SOUL_ESOWARD.sound()
-            }
-            // Pigman Sword
-            event.soundName == "mob.zombiepig.zpigangry" && event.pitch == 2.0f && event.volume == 0.3f -> {
-                ItemAbility.PIGMAN_SWORD.sound()
-            }
-            // Ember Rod
-            event.soundName == "mob.ghast.fireball" && event.pitch == 1.0f && event.volume == 0.3f -> {
-                ItemAbility.EMBER_ROD.sound()
-            }
-            // Fire Freeze Staff
-            event.soundName == "mob.guardian.elder.idle" && event.pitch == 2.0f && event.volume == 0.2f -> {
-                ItemAbility.FIRE_FREEZE_STAFF.sound()
-            }
-            // Staff of the Volcano
-            event.soundName == "random.explode" && event.pitch == 0.4920635f && event.volume == 0.5f -> {
-                ItemAbility.STAFF_OF_THE_VOLCANO.sound()
-            }
-            // Staff of the Volcano
-            event.soundName == "random.eat" && event.pitch == 1.0f && event.volume == 1.0f -> {
-                ItemAbility.STAFF_OF_THE_VOLCANO.sound()
-            }
-            // Holy Ice
-            event.soundName == "random.drink" && event.pitch.round(1) == 1.8f && event.volume == 1.0f -> {
-                ItemAbility.HOLY_ICE.sound()
-            }
+        for (ability in itemInHand.getActiveAbilities()) {
+            ability.onClick(event.clickType)
         }
     }
 
-    @SubscribeEvent
-    fun onItemClick(event: ItemClickEvent) {
-        if (AshfangFreezeCooldown.iscurrentlyFrozen()) return
-        handleItemClick(event.itemInHand)
-    }
+    private fun ItemStack.getActiveAbilities(): List<ActiveAbility> {
```

Comment:
Do not like this function. Since it is a called regular (every click) and checks over every Internalname in every Ability. This could be fixed with caching the value or reorganizing the data.

Reply:
cache added.